### PR TITLE
Findings cleaning and fixes

### PR DESCRIPTION
**Pull request by orouz**
follow up on https://github.com/build-security/kibana/pull/106 

included:
- [x] fix redundant computation caused by effect firing every render 
- [x] reduce code and and simplify naming

### CODE REVIEW

File: x-pack/plugins/cloud_security_posture/common/utils/helpers.ts [hunk 1]
```diff
@@ -14,12 +14,9 @@ import * as t from 'io-ts';
 export const isNonNullable = <T extends unknown>(v: T): v is NonNullable<T> =>
   v !== null && v !== undefined;
 
-export const getErrorOrUnknown = (e: unknown) =>
-  e instanceof Error ? e : new Error('Unknown Error');
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/common/utils/helpers.ts:18]: not needed

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx [hunk 2]
```diff
@@ -122,24 +61,27 @@ const mapEsQuerySortKey = (sort: readonly EsQuerySortValue[]): EsQuerySortValue[
...
+  const findingsResult = useFindings(dataView, findingsQuery, urlKey);
 
   return (
     <div data-test-subj={TEST_SUBJECTS.FINDINGS_CONTAINER}>
-      <FindingsSearchBar dataView={dataView} setQuery={setUrlQuery} {...urlQuery} {...fetchState} />
+      <FindingsSearchBar
+        dataView={dataView}
+        setQuery={setUrlQuery}
+        {...findingsQuery}
+        {...findingsResult}
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx:77]: this is pretty dope IMO in terms of readability 
types are `CspFindingsRequest` and `CspFindingsResponse` too

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx [hunk 3]
```diff
@@ -122,24 +61,27 @@ const mapEsQuerySortKey = (sort: readonly EsQuerySortValue[]): EsQuerySortValue[
...
   const { key: urlKey, urlQuery, setUrlQuery } = useUrlQuery(getDefaultQuery);
-  const findingsQuery = useMemo(
+  const findingsQuery: CspFindingsRequest = useMemo(
     () => ({ ...urlQuery, sort: mapEsQuerySortKey(urlQuery.sort) }),
     [urlQuery]
   );
 
-  const searchRequest = useFindings(dataView, findingsQuery, urlKey);
-
-  const fetchState = useMemo(() => getFetchState(searchRequest), [searchRequest]);
```
Comment (orouz) [inline x-pack/plugins/cloud_security_posture/public/pages/findings/findings_container.tsx:132]: `searchRequest` wasn't stable so isn't suited for a dependency prop. turns out we can use `select` from react-query to accomplish the same thing done here but nicer.

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx [hunk 4]
```diff
@@ -16,22 +16,22 @@ import {
...
 import { CspEvaluationBadge } from '../../components/csp_evaluation_badge';
-import type { FindingsUrlQuery, FindingsFetchState } from './findings_container';
+import type { CspFindingsRequest, CspFindingsResponse } from './use_findings';
 import { SortDirection } from '../../../../../../src/plugins/data/common';
 
-type TableQueryProps = Pick<FindingsUrlQuery, 'sort' | 'from' | 'size'>;
+type TableQueryProps = Pick<CspFindingsRequest, 'sort' | 'from' | 'size'>;
+type FindingsTableProps = CspFindingsResponse & BaseFindingsTableProps;
 
 interface BaseFindingsTableProps extends TableQueryProps {
```
Comment (ari-aviran) [inline x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx:30]: Consider swapping the order of these 2 definitions for logical reading order

**Review by ari-aviran — DISMISSED** [review]

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx (hunk 4 above)
Comment (orouz) [reply x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx:31]: do you mean like this: 
```typescript
type TableQueryProps = Pick<CspFindingsRequest, 'sort' | 'from' | 'size'>;

interface BaseFindingsTableProps extends TableQueryProps {
  setQuery(query: Partial<TableQueryProps>): void;
}

type FindingsTableProps = CspFindingsResponse & BaseFindingsTableProps;

```

tbh i wasn't sure about this 😄

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_search_bar.tsx [hunk 5]
```diff
@@ -7,19 +7,18 @@
...
+import type { CspFindingsRequest, CspFindingsResponse } from './use_findings';
 import type { CspClientPluginStartDeps } from '../../types';
 import { PLUGIN_NAME } from '../../../common';
 import type { DataView } from '../../../../../../src/plugins/data/common';
 
-type SearchBarQueryProps = Pick<FindingsUrlQuery, 'query' | 'filters' | 'dateRange'>;
+type SearchBarQueryProps = Pick<CspFindingsRequest, 'query' | 'filters' | 'dateRange'>;
+type FindingsSearchBarProps = CspFindingsResponse & BaseFindingsSearchBarProps;
 
 interface BaseFindingsSearchBarProps extends SearchBarQueryProps {
```
[the remark this answers is not in the document]
Comment (orouz) [reply x-pack/plugins/cloud_security_posture/public/pages/findings/findings_search_bar.tsx]: fixed

File: x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx (hunk 4 above)
Comment (orouz) [reply x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx]: fixed, they read logically (by order) now

Comment (ari-aviran) [reply x-pack/plugins/cloud_security_posture/public/pages/findings/findings_table.tsx]: Yes :)

**Review by ari-aviran — APPROVED** [review]

**Pull request merged**